src/views/AcademicCheckIn/index.js

```diff
@@ -169,24 +168,27 @@ const AcademicCheckIn = (props) => {
     if (props.authentication) {
       loadData();
     }
-
   }, [props.authentication, loading]);
 
+  useEffect(() => {
+    props.history.replace('/AcademicCheckIn', { step: activeStep });
+  }, [activeStep, props.history]);
+
   const handleNext = () => {
+    props.history.push('/AcademicCheckIn', { step: activeStep });
     setActiveStep((nextStep) => nextStep + 1);
-    window.history.pushState(activeStep, null, '');
+    props.history.replace('/AcademicCheckIn', { step: activeStep });
```

Comment:
This is a little bit confusing, but I suspect it may not behave as you expect. It looks like you're trying to push the current active step, and then replace that entry with the next active step (i.e. what is set in `setActiveStep`).

However, `set...` does not run immediately, but rather queues a state change to be executed on React's next render. So if you're depending on `activeStep` to have updated based on the call to `setActiveStep` here, then you are creating a race condition.